src/dbnode/ts/segment.go

```diff
@@ -25,19 +25,21 @@ import (
 
 	"github.com/m3db/m3/src/x/checked"
 	"github.com/m3db/m3/src/x/pool"
+
+	"github.com/m3db/stackadler32"
 )
 
 // Segment represents a binary blob consisting of two byte slices and
 // declares whether they should be finalized when the segment is finalized.
 type Segment struct {
 	// Head is the head of the segment.
 	Head checked.Bytes
-
 	// Tail is the tail of the segment.
 	Tail checked.Bytes
-
 	// SegmentFlags declares whether to finalize when finalizing the segment.
 	Flags SegmentFlags
+	// Checksum is the checksum for the segment.
+	Checksum uint32
```

Comment:
As discussed, perhaps let's make this non-exportable so you can't accidentally get a checksum that hasn't been calculated.

Then we can also remove `NewSegmentWithGeneratedChecksum` so it's just the single constructor and one way of accessing the checksum.